
\chapter{Automating PPR discovery} 
\label{chap:methods}

A brief overview of extraction and comparison

\section{pyHMMER}
\label{ssec:pyHMMER}

The HMMER suite provides all of the basic algorithms required in order to
perform an HMM search on a target amino acid sequence, but it does have some
limitations.

The first is that it is a command line program and does not have bindings to
any programming language.
HMMER reads inputs from files, and writes out tabular output data to file which
would be very time consuming to parse by hand.

HMMER can only compare a protein model with a protein
target, and so the genome must be translated before it can be searched.
There are a total of six possible reading frames (3 forwards and 3 backwards, 
due to the 3:1 nature of translation) and the genome must be searched in each
of these six frames.

HMMER is not commercial software and is developed by a group of scientists at
the Howard Hughes Medical Institute (HHMI) under an open licence for research
purposes.
As such, it contains a number of minor bugs which can sometimes cause problems
under particular circumstances, the most problematic of which causes enormous
memory usage (over 20GB\footnote{One particular instance of this error is due
to an unsigned integer wrap-around which causes significantly more memory to be
requested from the operating system than could possibly be needed}) and
prevents the program from completing.

In order to overcome these issues, a python wrapper for HMMER called
pyHMMER was designed and written.
Python was chosen as the main language for this project mainly because of its
excellent library support -- for example the biopython library solved many of 
the difficulties when working with biological sequences without extra effort.

pyHMMER does not implement all the features available in HMMER, but rather it
implements those which were most vital to this project.
Its main features are -
\begin{itemize}
  \item Read and write \emph{.hmm} files, HMMER's custom file format for
    storing HMMs
  \item Execute searches using 
    \emph{hmmsearch} and \emph{jackhmmer}, accepting all valid command line
    arguments and returning their output as biopython objects, 
    handling the creation and removal of all the necessary 
    temporary files automatically
  \item Seamlessly perform six-frame translations on the fly (implemented in C
    for best performance)
    and map the returned matches to the correct location
  \item Automatically terminate HMMER processes which attempt to allocate more
    memory than the system can sensibly be expected to provide and then call
    HMMER sequentially with subsections of the target\footnote{Linux only}
  \item Fully unit-tested with python's \emph{unittest} framework
\end{itemize}
pyHMMER has been developed under an open-source licence and is freely available
from \emph{https://github.com/haydnKing/pyHMMER}, although all code used in
this project was written by myself.


\section{Automated PPR Detection and Extraction}
\label{sec:ppr_extraction}

Several algorithms were developed and compared in order to extract PPRs from
unannotated genomes.
The results of each algorithm were compared with experimentally validated PPRs
and the best chosen.

Before development could begin, a HMM of the PPR repeat motif was required.
There are four such models available in
Pfam\footnote{http://pfam.sanger.ac.uk/search/keyword?query=PPR}, and each one
was tested on known PPRs in order to discover which model worked best when
searching for motifs.
It was found the PPR\_3 model is most sensitive to the motifs, and still
returns relatively few false positives.

Armed with this model, the final algorithm for discovering PPRs proceeds as
follows for each chromosome within the genome
\begin{enumerate}
  \item Perform a HMM search on the whole sequence. This will discover the most
    obvious PPRs only
  \item Group the motifs into clusters such that motifs which are on the same
    strand and are within a certain distance are put in the same cluster
  \item For each group, extract and `envelope' region containing it with large
    margins either side. If the envelope is on the reverse strand, the reverse
    complement should be taken such that all envelopes read in the forward
    direction
  \item \label{alg:envelopes}
    Search each envelope region for PPR motifs. This search is more
    focussed than the previous one and will reveal more motifs than previously.
    Discard any envelopes which only contain one motif.
  \item Starting from the first position of the first motif, search backwards
    one codon at a time until a start codon (`ATG') is reached
  \item Starting from the last position in the final motif, search forwards one
    codon at a time until a stop codon is reached (`TGA', `TAG' or `TAA'). This
    fixes the location of the PPR
  \item Check for PPRs which overlap -- this indicates that two motifs which
    were believed to belong to separate proteins in fact belong to the same
    one. Each set of overlapping proteins should be removed and a new, larger
    envelope extracted. The algorithm then continues again from step
    \ref{alg:envelopes}
  \item Check for PPRs where the motifs have filled the envelopes -- i.e. ones
    which are missing a start or a stop codon due to not having searched far
    enough. Extract larger envelopes for these proteins and continue from step
    \ref{alg:envelopes}
  \item Search each protein for gaps between motifs which are the correct size
    to fit a PPR motif. Search these regions specifically, increasing HMMER's
    sensitivity, (using the F1, F2 and F3 arguments) to look for reluctant PPR
    motifs. Also search the beginning and the end of the protein in this way
  \item Search each protein for small (2/3 codon) gaps between the motifs and
    move the end position of the previous motif in order to fill these gaps.
    This allows the motifs to be classified as P, L or S
  \item Classify the proteins depending on which types of motifs they contain.
    Extract the protein sequence of each tail sequence and classify it using
    \emph{jackhmmer} to search for the known consensus sequences for E, E+ and
    DYW motifs
  \item Predict each protein's sub-cellular location using the \emph{targetP}
    program
\end{enumerate}



\section{Predicting PPR Binding regions}
\label{sec:ppr_binding_prediction}

Introduce the problem and the data available.

\subsection{Direct HMMs}
\label{sec:hmm_binding}

Explain this method and why it failed

\subsection{PSSM}
\label{sec:pssm_binding}

Explain why this method was more successful, but why it fails to recognise a
precise binding region and how this problem was overcome.


